src/server/game/Entities/Player/PlayerMisc.cpp

```diff
@@ -321,6 +321,10 @@ void Player::ResetInstances(ObjectGuid guid, uint8 method, bool isRaid)
         }
             break;
     }
+
+    Player* player = ObjectAccessor::FindConnectedPlayer(guid);
+
+    sScriptMgr->OnPlayerResetsInstances(player, method, isRaid);
```

Comment:
Pass the guid and get the player inside the hook if needed

Reply:
Actually, we're inside the player class, so pass **this** instead of guid

Reply:
when i was creating it i wanted to get player but sure i agree